eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+import java.util.regex.Pattern;
+
+/**
+ * Requirement of Javadoc annotation @codesnippet check:
+ *  (1) Use '{@codesnippet ...}' instead of '<code>', '<pre>'.
+ *  (2) No multiple lines span at the '{@code ...}'.
```

Comment:
What does this mean?

Reply:
Suggestion: `{@literal {@code ...}}` tags do not span multiple lines.

Reply:
{@code ...} in a single line is valid but {@code avccsscscss
cscacasc} is not valid